### PR TITLE
Make the `@yamada-ui/charts` component headless

### LINKED ISSUE #1858: Make the `@yamada-ui/charts` component headless
script: make-component-headless
package: @yamada-ui/charts

### Description
Currently, the `@yamada-ui/charts` component has minimal styles set within the component's source code.
We want to make this style transferable to the theme and make the component headless, enabling more flexible UI customization.

### Problem Statement/Justification
By making the `@yamada-ui/charts` component headless, we can achieve more flexible styling tailored to the project.

### Proposed Solution or API
The styles within the component are mostly defined by `css: CSSUIObject` and set to a prop named `__css`. We will transfer these styles to the `packages/theme/src/components`.

The code below is just an example and not the actual code.

`packages/components/input/src/input-group.tsx`
```diff
const css: CSSUIObject = {
- width: "100%",
- display: "flex",
- position: "relative",
  ...styles.container,
}
```

`packages/theme/src/components/input.ts`
```diff
baseStyle: {
  container: {
+   width: "100%",
+   display: "flex",
+   position: "relative",
  },
  field: {
    width: "100%",
    minWidth: 0,
    outline: 0,
    position: "relative",
    appearance: "none",
    transitionProperty: "common",
    transitionDuration: "normal",
    _disabled: {
      opacity: 0.4,
      cursor: "not-allowed",
    },
  },
  element: {
    color: ["blackAlpha.600", "whiteAlpha.700"],
  },
},
```

> [!TIP]
>
> In cases where the styles within the component are dynamically changed, there may be instances where it is not possible to transfer the styles to the `theme`.
> In such cases, it needs to be discussed in this issue. Please feel free to send comments to this issue.

### Additional Information
The `@yamada-ui/charts` component exists in `packages/components/charts/src`. You need to check all the components in this folder and if they contain styles internally, you need to modify them.

This change could have a significant impact on projects using Yamada UI, so we will work on it within the scope of `v2.0`.
Please create the branch for working on this issue from the remote branch `v2.0`, and direct your pull requests to the `v2.0` branch as well.

Please select `major` for the changeset and create a pull request.

### PR DESCRIPTION
**Pull request by EGASHIRAAkihide**
[Refactor] Closes #1858 

## Description

- Make the `@yamada-ui/charts` component headless
  - LineChart
  - BarChart
  - AreaChart
  - RadarChart
  - DonutChart
  - PieChart

### CODE REVIEW

File: .changeset/witty-adults-sin.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/charts": major
```
Comment (zonoryo17) [inline .changeset/witty-adults-sin.md:2]: I think you need to add “@yamada-ui/theme”.

Comment (EGASHIRAAkihide) [reply .changeset/witty-adults-sin.md]: @zonoryo17 

Thank you for your advice.
I've fixed and updated that.

**Review by koralle — DISMISSED** [review]

**Review by zonoryo17 — DISMISSED** [review]

**Review by 108yen — CHANGES_REQUESTED** [review]
All chart components inherit the `LineChart` theme.
Only the `line-chart.ts` theme file should be changed for the `styles.container`.

Could you also move the following styles to the theme file? (There are others.)
https://github.com/yamada-ui/yamada-ui/blob/081a7798b5b3cd11e0d592e2e403e401e42fe681/packages/components/charts/src/use-area-chart.ts#L212-L217

**Review by 108yen — CHANGES_REQUESTED** [review]
Only styles common to all charts, such as `styles.container`, can be integrated into `line-chart.ts`.
For other styles, you need to consider whether they are common or not, and then decide where to put them.
(If you are not sure, please ask.)

I used the `use-area-chart.ts` change as an example, but other charts should be scrutinized as well.

This change is quite difficult.

**Comment by hirotomoyamada** [comment]
@EGASHIRAAkihide 

Hi, thanks for the PR.

Do not include `pnpm-lock.yaml` in this pull request.

**Comment by EGASHIRAAkihide** [comment]
Im pushing the updated one today.

**Comment by EGASHIRAAkihide** [comment]
@hirotomoyamada 

To solve not to change `pnpm-lock.yaml`, I've tried to fix it.

A. Remove `node_modules` folder, `pnpm-lock.yaml`. Then, command `$pnpm install` to install them again.

----------

B. Copy the original `pnpm-lock.yaml` file, paste to this PR's file.

Those didn't work.

Could you help me how to resolve this problem?

**Review by 108yen — CHANGES_REQUESTED** [review]
@EGASHIRAAkihide 
Fixed `pnpm-lock.yml`.

You have modified only `use-area-chart.ts`, can you modify the other charts as well?

**Review by koralle — DISMISSED** [review]

**Review by 108yen — APPROVED** [review]
Thank you!

**Review by zonoryo17 — APPROVED** [review]
Thank you for addressing the issue.

**Comment by hirotomoyamada** [comment]
Thanks!

**Pull request merged**